src/separateLater/Sentinel1/FileIO.jl

```diff
@@ -12,11 +12,46 @@ end
 
 
 """
-readSwathSLC(filepath::String)
-Returns: ::SwathSLC
+    readTiff(filepath::String, window=nothing, convertToDouble = true)
+
+    Read a Sentinel 1 tiff file.
+    # Examples:
+    ```jldoctest
+    julia> filepath = "s1a-iw3-slc-vv-20220918t074921-20220918t074946-045056-056232-006.tiff"
+    julia> data = readSwathSLC(filePath, [(501,600),(501,650)]);
+    julia> typeof(data)
+    Matrix{ComplexF64}
+    julia> size(data)
+    (100,150)
+    ```
 """
-function readSwathSLC(filepath::String, window)
-    error("Not implemented")
+function readTiff(filepath::String, window=nothing, convertToDouble = true)
+
+    dataset = ArchGDAL.readraster(filepath)
+
+    if window[1][2] > ArchGDAL.width(dataset)
+        @warn "The window exceeds the dataset width $(ArchGDAL.width(dataset))."
+    end
+    if window[2][2] > ArchGDAL.height(dataset)
+        @warn "The window exceeds the dataset height $(ArchGDAL.height(dataset))."
+    end
+
+    if !isnothing(window)
```

Comment:
Minor thing: is there a reason you didn't flip this like
`if isnothing(window)
dataset = dataset[:, :, 1]
else
dataset = dataset[window]`
Double negations like `!isnothing()` is a bit hard for me to understand, but maybe there is some reason.